src/hist/basehist.py

```diff
@@ -401,20 +403,45 @@ def plot2d_full(
 
         return hist.plot.plot2d_full(self, ax_dict=ax_dict, **kwargs)
 
+    def plot_ratio(
+        self,
+        other: Callable[[np.ndarray], np.ndarray],
+        *,
+        ax_dict: "Optional[Dict[str, matplotlib.axes.Axes]]" = None,
+        **kwargs: Any,
+    ) -> "Tuple[MainAxisArtists, RatiolikeArtists]":
```

Comment:
Totally not on you, but I'm pretty sure none of these quotes are needed. They should all go away anyway when we go to 3.7+.

Reply:
Hm. With the following diff

```diff
diff --git a/src/hist/basehist.py b/src/hist/basehist.py
index 45c1665..21021d5 100644
--- a/src/hist/basehist.py
+++ b/src/hist/basehist.py
@@ -409,7 +409,7 @@ class BaseHist(bh.Histogram, metaclass=MetaConstructor, family=hist):
         *,
         ax_dict: "Optional[Dict[str, matplotlib.axes.Axes]]" = None,
         **kwargs: Any,
-    ) -> "Tuple[MainAxisArtists, RatiolikeArtists]":
+    ) -> Tuple[MainAxisArtists, RatiolikeArtists]:
         """
         ``plot_ratio`` method for ``BaseHist`` object.
```

my posted example fails with

```pytb
$ python example.py 
Traceback (most recent call last):
  File "example.py", line 4, in <module>
    import hist
  File "/home/feickert/Code/GitHub/Scikit-HEP/hist/src/hist/__init__.py", line 11, in <module>
    from .basehist import BaseHist
  File "/home/feickert/Code/GitHub/Scikit-HEP/hist/src/hist/basehist.py", line 62, in <module>
    class BaseHist(bh.Histogram, metaclass=MetaConstructor, family=hist):
  File "/home/feickert/Code/GitHub/Scikit-HEP/hist/src/hist/basehist.py", line 412, in BaseHist
    ) -> Tuple[MainAxisArtists, RatiolikeArtists]:
NameError: name 'MainAxisArtists' is not defined
```

Reply:
Ahh, it's because of the TYPE_CHECKING line above. Not sure if that's required (except for ellipsis, that's required), but that's why.

Reply:
Yeah, I just assumed as there was

```python
    from mplhep.plot import Hist1DArtists, Hist2DArtists
```

under it as well that I should keep all types from plotting in there as well. I can move them out if that would be better.

Reply:
We can address later, I'm guessing this might be due to an older version missing those? Soon 3.4+ will be required, I expect, due to mplhep.